cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -83,12 +85,16 @@ public DatasetTypeManager(CConfiguration cConf, MDSDatasetsRegistry mdsDatasets,
     this.locationFactory = locationFactory;
     this.defaultModules = Maps.newLinkedHashMap(defaultModules);
     this.allowDatasetUncheckedUpgrade = cConf.getBoolean(Constants.Dataset.DATASET_UNCHECKED_UPGRADE);
+    this.extensionModules = getExtensionModules(this.cConf);
   }
 
   @Override
   protected void startUp() throws Exception {
     deleteSystemModules();
     deployDefaultModules();
+    if (extensionModules != null && extensionModules.size() > 0) {
```

Comment:
I don't think `extensionModules` will be null, since `getExtensionModules` never returns null. Also, use `!isEmpty()` instead of `size() > 0`.